sdk/eventhub/Azure.Messaging.EventHubs/api/Azure.Messaging.EventHubs.netstandard2.0.cs

```diff
@@ -244,6 +244,29 @@ public partial class EventProcessorCheckpoint
         public string PartitionId { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
         public long? SequenceNumber { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
     }
+    public partial class EventProcessorOptions
+    {
+        public EventProcessorOptions() { }
+        public Azure.Messaging.EventHubs.EventHubConnectionOptions ConnectionOptions { get { throw null; } set { } }
+        public Azure.Messaging.EventHubs.Consumer.EventPosition DefaultStartingPosition { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
```

Comment:
Because developers are responsible for returning the list of checkpoints, we're advising them to return a checkpoint with the default for any partition that they want to have different from the common default.   The `ListCheckpointsAsync` method has:

```
        /// <remarks>
        ///   Should a partition not have a corresponding checkpoint, the <see cref="EventProcessorOptions.DefaultStartingPosition" /> will
        ///   be used to initialize the partition for processing.
        ///
        ///   In the event that a custom starting point is desired for a single partition, or each partition should start at a unique place,
        ///   it is recommended that this method express that intent by returning checkpoints for those partitions with the desired custom
        ///   starting location set.
        /// </remarks>
```

Reply:
We considered a bunch of other ideas, including having a dictionary in the options, but since they've got full control over telling us about checkpoints, it seemed to be confusing to have multiple ways to do that.

Should we think about having a remark similar to the above on that `DefaultStartingPosition` member?

Reply:
The checkpoint logic requires that I return either an Offset or SequenceNumber.  How do I convert `EventPosition.Latest` or `EventPosition.Earliest` or `EventPosition.FromDate(DateTmeOffset.UTC.Subtract(TimeSpan.FromHours(1))`? to sequence numbers or offsets?  Is this even possible?

Reply:
It's not; the closest that you'd get is to set the default position to earliest or latest and then returning checkpoint data for things where the position is known.   Looks like a gap would be having a mix of Earliest and Latest at the same time.   

I wonder if we should change the checkpoint to instead just return an Event Position and let callers be responsible for figuring out what that should be.

Reply:
> I wonder if we should change the checkpoint to instead just return an Event Position and let callers be responsible for figuring out what that should be.

I was thinking about something like these lines as well.  That also helps with the "the checkpoint object contains two different ways of representing the "same thing" (offset vs sequence number) which we felt a little bad about in the past".

Reply:
I just pitched it to Krzysztof.   Assuming he has no objections, I'll make the adjustment.

Reply:
I also added the remarks on the options, which hopefully fills in the gap so that its understandable from both places.